lib/RequestRouter.js

```diff
@@ -514,8 +520,8 @@ RequestRouter.prototype.addPropertyError = function(property, errorCode) {
         });
     }
 
-    if (this._request.getMethod() == Request.METHOD_POST && this.isCollection()) {
-        //remove the appended items@ if posting to a collection
+    if (!!~[Request.METHOD_PATCH, Request.METHOD_POST, Request.METHOD_PUT, Request.METHOD_DELETE].indexOf(this.getMethod()) && this.isCollection()) {
```

Comment:
If you want to patch an individual item in a collection, you would use the uri of that specific item. For example, to create a new User I would `POST` a single object to `/users`, which is normally a collection endpoint. To update the user, we would make a `PATCH` request to that object's URL, e.g. `/users/123`, which is not a collection endpoint.

It really comes down to `POST` not being idempotent, while the other methods are. So we need to handle it differently.